### PR TITLE
Embed the Custom and Modifiable Object Entry's permissions in the nestedFields parameter for POST, PUT and PUT by ERC request

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://liferay.atlassian.net/browse/LPD-31544
https://liferay.atlassian.net/browse/LPD-31545
https://liferay.atlassian.net/browse/LPD-31921

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/util/ServiceContextUtil.java [hunk 1]
```diff
@@ -22,12 +23,14 @@
 public class ServiceContextUtil {
 
 	public static ServiceContext createServiceContext(
-		Locale locale, ObjectEntry objectEntry, long userId) {
+		Locale locale, ObjectEntry objectEntry, long userId,
+		ModelPermissions modelPermissions) {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/util/ServiceContextUtil.java:27]: Is this order correct?
What was the pattern that you followed?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -1578,6 +1588,9 @@ public ObjectEntry updateObjectEntry(
 			ObjectEntryThreadLocal.setSkipObjectValidationRules(false);
 		}
 
+		_updateResourcePermissions(
+			objectDefinition, objectEntry, serviceContext);
+
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1593]: I'm wondering about the order in which this method is called.
In the ADD method, you call this method after` _updateAsset(...)` and here it is before.
Is there a required order for these methods to work?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java [hunk 3]
```diff
@@ -7612,6 +7613,128 @@ public void testPostObjectEntryWithKeywordsAndTaxonomyCategoryIdsWhenCategorizat
 		}
 	}
 
+	@Test
+	public void testPostPutCustomObjectEntryWithPermissions() throws Exception {
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:7617]: Did you find a pattern that tests Post and Put using the same test method and that uses this prefix `testPostPut`?
I prefer to separate them.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1593]: @marcelabc It was my mistake while fixing the rebase conflict. I'll change it to put it before `_updateAsset`

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java (hunk 3 above)
Comment (carolmariaabb) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:7617]: @marcelabc I saw other methods in this same test class following this pattern:

Example: [testPatchPutCustomObjectEntryExternalReferenceCode](https://github.com/liferay-objects/liferay-portal/blob/db30ccde22468d67701614129f79463b9dd2f430/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java#L6879)

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/util/ServiceContextUtil.java (hunk 1 above)
Comment (carolmariaabb) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/util/ServiceContextUtil.java]: @marcelabc I updated the code

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java (hunk 3 above)
Comment (marcelabc) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:7617]: Thank you!

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
https://github.com/liferay-headless/liferay-portal/pull/2161

**Pull request closed without merging**